### PR TITLE
Use suspend-on-failure for streaming jobs by default [HZ-1648]

### PR DESCRIPTION
**Pull request by Fly-Style**

### CODE REVIEW

**Comment by Fly-Style** [comment]
run-lab-run

**Review by ivanthescientist — APPROVED** [review]

File: hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/PlanExecutor.java [hunk 1]
```diff
@@ -188,10 +188,14 @@ SqlResult execute(CreateIndexPlan plan) {
 
     SqlResult execute(CreateJobPlan plan, List<Object> arguments) {
         List<Object> args = prepareArguments(plan.getParameterMetadata(), arguments);
+        boolean isStreamingJob = plan.isInfiniteRows();
         JobConfig jobConfig = plan.getJobConfig()
                 .setArgument(SQL_ARGUMENTS_KEY_NAME, args)
                 .setArgument(KEY_SQL_QUERY_TEXT, plan.getQuery())
-                .setArgument(KEY_SQL_UNBOUNDED, plan.isInfiniteRows());
+                .setArgument(KEY_SQL_UNBOUNDED, isStreamingJob);
```
Comment (krzysztofslusarski) [inline hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/PlanExecutor.java:195]: What about the other plans? Select/DML? Those also can be streaming.

Comment (k-jamroz) [reply hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/PlanExecutor.java:195]: Is there any description of this change apart from issue title?

Do we want one-shot streaming SQL queries (that is not executed using `CREATE JOB`) to be suspended? From the user point of view such query probably failed. I am not sure if they would be able to "reconnect" to it after restart to get more select results. For DML that might make more sense, but user would not know/expect that failed INSERT/UPDATE is still sleeping somewhere to be resumed. Some would also have to clean snapshots.

All-in-all, for me it makes sense to enable `suspend-on-failure` only for cases when user explicitly wanted something more durable, i.e. used `CREATE JOB`.

There is also a technical reason: normal queries are light jobs which cannot be suspended.

File: hazelcast-sql/src/test/java/com/hazelcast/jet/sql/SqlMetadataInJobConfigTest.java [hunk 2]
```diff
@@ -103,6 +106,31 @@ public void test_createJobMetadata() {
         JobConfig config = completedJobs.get(0).getConfig();
         assertEquals(sql, config.getArgument(KEY_SQL_QUERY_TEXT));
         assertEquals(Boolean.FALSE, config.getArgument(KEY_SQL_UNBOUNDED));
+        assertEquals(Boolean.FALSE, config.isSuspendOnFailure());
+    }
+    @Test
```
Comment (k-jamroz) [inline hazelcast-sql/src/test/java/com/hazelcast/jet/sql/SqlMetadataInJobConfigTest.java:111]:
```suggestion

    @Test
```

**Review by k-jamroz — APPROVED** [review]

File: hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/PlanExecutor.java (hunk 1 above)
Comment (Fly-Style) [reply hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/PlanExecutor.java:195]: @krzysztofslusarski it's applied for CREATE JOB statement, and it should work fine for any plans covered by `CreateJobPlan`

Comment (Fly-Style) [reply hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/PlanExecutor.java:195]: @k-jamroz [link](https://hazelcast.atlassian.net/wiki/spaces/PM/pages/4318986241/Jet+Job+Management+Improvements)

**Comment by Fly-Style** [comment]
run-sql-only

**Review by viliam-durina — APPROVED** [review]

**Review by krzysztofslusarski — APPROVED** [review]

**Pull request merged**